api/src/test/java/com/epam/pipeline/controller/billing/BillingControllerTest.java

```diff
@@ -0,0 +1,179 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = BillingController.class)
+public class BillingControllerTest extends AbstractControllerTest {
+
+    private static final long COST = 7367L;
+    private static final String BILLING_URL = SERVLET_PATH + "/billing";
+    private static final String GET_BILLING_CHART_INFO_URL = BILLING_URL + "/charts";
+    private static final String GET_BILLING_CHART_INFO_PAGINATED_URL = GET_BILLING_CHART_INFO_URL + "/pagination";
+    private static final String GET_BILLING_CENTERS = BILLING_URL + "/centers";
+    private static final String REQUEST_JSON = "{\"from\":\"-999999999-01-01\"," +
+                                                "\"to\":\"+999999999-12-31\"," +
+                                                "\"filters\":{\"test\":[\"test\"]}," +
+                                                "\"interval\":\"1d\"," +
+                                                "\"grouping\":\"BILLING_CENTER\"," +
+                                                "\"loadDetails\":true," +
+                                                "\"pageSize\":5," +
+                                                "\"pageNum\":1}";
+    private BillingChartRequest billingChartRequest;
+    private List<BillingChartInfo> billingChartInfos;
+
+    @Autowired
+    private BillingApiService mockBillingApiService;
+
+    @Before
+    public void setUp() {
+        final BillingChartInfo billingChartInfo = BillingChartInfo.builder()
+                .cost(COST)
+                .build();
+        billingChartRequest = new BillingChartRequest(
+                LocalDate.MIN, LocalDate.MAX, Collections.singletonMap("test", Collections.singletonList("test")),
+                DateHistogramInterval.DAY, BillingGrouping.BILLING_CENTER, true, 5L, 1L
+        );
+
+        billingChartInfos = Collections.singletonList(billingChartInfo);
+    }
+
+    @Test
+    public void shouldFailGetBillingChartInfoForUnauthorizedUser() throws Exception {
+        mvc().perform(post(GET_BILLING_CHART_INFO_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnBillingChartInfo() throws Exception {
+        Mockito.doReturn(billingChartInfos).when(mockBillingApiService).getBillingChartInfo(billingChartRequest);
+
+        final MvcResult mvcResult = mvc().perform(post(GET_BILLING_CHART_INFO_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(REQUEST_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockBillingApiService).getBillingChartInfo(billingChartRequest);
+        final ArgumentCaptor<BillingChartRequest> billingChartRequestCaptor =
```

Comment:
To my mind capture check is unnecessary here: you have already add mock call verification with argument. `Mockito#verify` would be enough.